main.tf

```
@@ -2,25 +2,64 @@
 
 ## Exemplo
 
-# provider "google" {
-#   project = var.project
-#   region  = var.region
-# }
-
-# module "groundwork" {
-#   source   = "github.com/mentoriaiac/iac-modulo-groundwork.git?ref=v0.1.0"
-#   project  = var.project
-#   vpc_name = "groundwork"
-# }
-
-# variable "project" {
-#   description = "Nome do projeto (Default é staging)"
-#   type        = string
-#   default     = "direct-link-325016"
-# }
-
-# variable "region" {
-#   description = "Nome da região"
-#   type        = string
-#   default     = "us-central1"
-# }
\ No newline at end of file
+provider "google" {
+  project = var.project
+  region  = var.region
+}
+
+module "network_gcp" {
+  source      = "github.com/Rehzende/iac-modulo-rede-gcp.git?ref=d8c3058"
+  project     = var.project
+  vpc_name    = "rede-mentoria"
+  direction   = "INGRESS"
+  target_tags = ["nomad"]
+  source_tags = ["nomad"]
+  subnetworks = [
+    {
+      name          = "subnet-nomad"
+      ip_cidr_range = "10.0.0.0/16"
+      region        = "us-central1"
+    }
+  ]
+
+  firewall_allow = [
+    {
+      protocol = "tcp"
+      port     = [22, 4646, 4647, 4648]
+    }
+  ]
+}
+
+module "nomad_servers" {
+  source = "github.com/mentoriaiac/iac-modulo-compute-gcp.git?ref=v0.1.0"
+
+  for_each = var.server_pool
+
+  project                 = var.project
+  instance_name           = each.value.instance_name
```

```suggestion
  instance_name           = each.key
```
Aqui poderia usar a chave direto, já que ela também é usada para identificar an instância.

---

Alterado!